algebra-core/src/serialize/mod.rs

```diff
@@ -322,6 +375,20 @@ macro_rules! impl_prime_field_serializer {
                 reader.read_exact(&mut masked_bytes[..output_byte_size])?;
                 Ok(Self::read(&masked_bytes[..])?)
             }
+
+            #[allow(unused_qualifications)]
+            fn deserialize_unchecked<R: crate::io::Read>(
+                reader: &mut R,
+            ) -> Result<Self, crate::serialize::SerializationError> {
+                const BYTE_SIZE: usize = $byte_size;
+
+                let (_, output_byte_size) =
+                    crate::serialize::buffer_bit_byte_size($field::<P>::size_in_bits());
+
+                let mut masked_bytes = [0; BYTE_SIZE];
+                reader.read_exact(&mut masked_bytes[..output_byte_size])?;
+                Ok(Self::read(&masked_bytes[..])?)
```

Comment:
This is duplicate code with the `deserialize` just above.
The default implementation will do the job and the implementation for `deserialize_unchecked` can be removed.